### DISCUSSION TITLE
DirectML Support?

### DISCUSSION THREAD

**Discussion by DK013** [openai/whisper#979] [Ideas]
I'm a beginer to python , otherwise I would do it myself.
Anybody has any idea if it's possible to use pytorch [DirectML plugin](https://pypi.org/project/pytorch-directml/) use any DirectX12 supported GPU for transcription instead of just relying on CUDA?
I found this [blog](https://learn.microsoft.com/en-us/windows/ai/directml/gpu-pytorch-windows) from Microsoft. In theory it seems possible but I have no idea how to do it myself.

**Comment by ultrasound1372** [comment]
A friend of mine just got a new computer, and it has AMD Radian, not NVIDIA. She wants to make use of Whisper to transcribe a significant portion of audio, no clouds for privacy, but is not the most tech-savvy, and would need to be able to run it on Windows. This would help a lot.

**Comment by TheBreadGuy** [comment]
https://github.com/Const-me/Whisper

windows ap with ui. seems to use DirectX12 on my 4090 it is very  fast

**Reply by DK013** [reply]
Great Find 🙏
This is a really excelent implementation...though it uses an old version of the whisper.cpp project.
The only problem with this library is the author didn't bother much with the real time transcription feature. The current implementation is bad and has really high latency
